python/lsst/pipe/base/graphBuilder.py

```diff
@@ -707,22 +707,37 @@ def resolveDatasetRefs(self, registry, collections, run, commonDataIds, *, skipE
                 # These may have dimensions that extend beyond those we queried
                 # for originally, because we want to permit those data ID
                 # values to differ across quanta and dataset types.
-                # For example, the same quantum may have a flat and bias with
-                # a different calibration_label, or a refcat with a skypix
-                # value that overlaps the quantum's data ID's region, but not
-                # the user expression used for the initial query.
                 for datasetType in task.prerequisites:
                     lookupFunction = lookupFunctions.get(datasetType.name)
                     if lookupFunction is not None:
+                        # PipelineTask has provided its own function to do the
+                        # lookup.  This always takes precedence.
                         refs = list(
                             lookupFunction(datasetType, registry, quantum.dataId, collections)
                         )
+                    elif (datasetType.isCalibration()
+                            and datasetType.dimensions <= quantum.dataId.graph
+                            and quantum.dataId.graph.temporal):
+                        # This is a master calibration lookup, which we have to
+                        # handle specially because the query system can't do a
+                        # temporal join on a non-dimension-based timespan yet.
+                        timespan = quantum.dataId.timespan
+                        try:
+                            refs = [registry.findDataset(datasetType, quantum.dataId,
```

Comment:
Can you be sure that this will only ever return one, is an exception raised if more than one could be found for a timespan? Does the certification process prevent this?

Reply:
Yes, `findDataset` will raise in that case.  The certification process is supposed to guarantee that is impossible for any calibration I can think of, but it's not something we can guarantee at the level of database constraints.  So the exception basically says, "someone probably put together a malformed calibration repo".